web/src/shared/components/document-pages/document-pages.tsx

```diff
@@ -142,54 +142,85 @@ const DocumentPages: React.FC<DocumentPagesProps> = ({
+                                            containerRef={containerRef}
+                                            editable
+                                            isShownAnnotation={true}
+                                            onAnnotationCopyPress={onAnnotationCopyPress}
+                                            onAnnotationCutPress={onAnnotationCutPress}
+                                            onAnnotationPastePress={onAnnotationPastePress}
+                                            onAnnotationUndoPress={onAnnotationUndoPress}
+                                            onAnnotationRedoPress={onAnnotationRedoPress}
+                                            onEmptyAreaClick={onEmptyAreaClick}
+                                        />
+                                    </Fragment>
+                                );
+                            })}
                         </ResizableSyncedContainer>
-                        {userPages.map(({ user_id, page_num }) => (
+
+                        {latestRevisionByAnnotators.map((revision) => (
                             <div
-                                key={user_id}
+                                key={revision.user_id}
                                 className={styles['additional-pages-with-user-name']}
                             >
-                                <SplitAnnotatorInfo annotatorName={getAnnotatorName(user_id)} />
+                                <SplitAnnotatorInfo
+                                    annotatorName={getAnnotatorName(revision.user_id)}
+                                />
                                 <SyncedContainer
                                     className={cx(
                                         styles['split-document-page'],
                                         styles['additional-page']
                                     )}
                                 >
-                                    <DocumentSinglePage
-                                        userId={user_id}
-                                        annotations={annotationsByUserId[user_id]}
-                                        scale={fullScale}
-                                        pageSize={apiPageSize}
-                                        pageNum={page_num}
-                                        onAnnotationSelected={(scaledAnn?: Annotation) =>
-                                            onSplitAnnotationSelected(fullScale, user_id, scaledAnn)
-                                        }
-                                    />
+                                    {pageNumbers.map((pageNum) => {
+                                        const isShowAnnotation =
+                                            !!latestRevisionByAnnotatorsWithBounds[
+                                                revision.user_id
+                                            ].filter((obj) => obj.pageNum === pageNum).length;
```

Comment:
you can use `.some` to check if there are any items, it returns `boolean`

Reply:
thanks, fixed https://github.com/epam/badgerdoc/pull/584/commits/5fc548e91e1f525470ad4a988b68acafeff71e18